File: R/8kyu/8kyu-function-1-hello-world.r
# Function 1 - hello world
# Language: R
# Difficulty: 8 kyu

# Solution 1
greet <- function(){
  print("hello world!")
}

# Shortest solution, can add return() for best practices
greet <- function() {
  "hello world!"
}


# Tests
test_that("Test for hello world", {
  expect_equal(greet(), "hello world!")
  expect_equal(paste("someone", greet(), sep=" "), "someone hello world!")
})

File: Julia/8kyu/8kyu-function-1-hello-world.jl
# Function 1 - hello world
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

# Tests

module Solution
  export greet

  function greet()
    return "hello world!"
  end

end


# Tests
using FactCheck
using .Solution

facts("Testing") do
  @fact greet() --> "hello world!"
end